app/src/main/assets/nodejs-project/DIAGNOSTICS.md

```
@@ -32,6 +32,23 @@ grep -i "429\|Too Many Requests\|rate.limit" node_debug.log | tail -10
 **Diagnosis:** Telegram rate limits: ~30 messages/second to different chats, ~20 messages/minute to same chat, ~1 message/second for same chat. Bulk sending or rapid tool status updates can trigger this.
 **Fix:** Reduce message frequency. Batch status updates into single messages. If persistent, wait 30-60 seconds before retrying. This is transient — no config change needed.
 
+### Rich Messages Falling Back to Classic Formatting (BAT-1050)
+**Symptoms:** Replies render as plain/HTML instead of rich (no tables/headings/math), or logs show `sendRichMessage` warnings.
+**Check:**
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)